### ISSUE TITLE
getorf failed

### ISSUE THREAD

**Issue by ting-hsuan-chen** [qjiangzhao/TEtrimmer#40]
Hi Jiangzhao,

Thank you for developing TEtrimmer!

I tried to run TEtrimmer using the test files you provided. However it was not successful...

First of all, among the three installation methods, only the singularity method worked for me. So I run TEtrimmer as follows:
```bash
sbatch << EOF
#!/bin/bash
#SBATCH -J $JOB
#SBATCH -o $JOB.%j.out
#SBATCH -e $JOB.%j.err
#SBATCH --cpus-per-task=$THREADS
#SBATCH --mem=$RAM
#SBATCH --time=$TIME

apptainer exec --writable-tmpfs \
$TOOL/tetrimmer_1.4.0--hdfd78af_0.sif \
TEtrimmer \
-i $OUT/test/test_input.fa \
-g $OUT/test/test_genome.fasta \
-o $OUT/test \
--pfam_dir $TOOL/pfam \
-t 20 \
--classify_all

EOF
```

Several intermediate files were generated, but some were missing, especially the final consensus.fasta file. So I looked into the log file $JOB.%j.out, and found lots of error messages, mainly about getorf. For example:
```text
getorf failed for rnd_6_family_3291.fasta with error code 1.

rnd_6_family_3291 is skipped due to blast hit number is 0


getorf failed for rnd_6_family_2561.fasta with error code 1.

getorf failed for rnd_1_family_587.fasta with error code 1.

getorf failed for rnd_1_family_667.fasta with error code 1.

getorf failed for rnd_5_family_182.fasta with error code 1.

getorf failed for rnd_1_family_588.fasta with error code 1.

getorf failed for rnd_1_family_188.fasta with error code 1.

getorf failed for rnd_1_family_470.fasta with error code 1.

getorf failed for rnd_1_family_667.fasta.b.bed_uf.bed_g_1.bed_fm.bed_0_0_bcln.fa_aln.fa_cl.fa_gs.fa_ce.fa_bc.fa_ce.fa_gs.fa_bc.fa_co.fa with error code 1.

ORF or PFAM prediction failed for rnd_1_family_667 with error:
can only concatenate str (not "bytes") to str

Traceback (most recent call last):
  File "/usr/local/share/tetrimmer/orfdomain.py", line 192, in run_getorf
    subprocess.run(get_orf_command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/usr/local/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['getorf', '-sequence', '../TEtrimmer/test/TEtrimmer_output_20240731_142908/Multiple_sequence_alignment/rnd_1_family_667.fasta.b.bed_uf.bed_g_1.bed_fm.bed_0_0_bcln.fa_aln.fa_cl.fa_gs.fa_ce.fa_bc.fa_ce.fa_gs.fa_bc.fa_co.fa', '-outseq', '../TEtrimmer/test/TEtrimmer_output_20240731_142908/Multiple_sequence_alignment/rnd_1_family_667.fasta.b.bed_uf.bed_g_1.bed_fm.bed_0_0_bcln.fa_aln.fa_cl.fa_gs.fa_ce.fa_bc.fa_ce.fa_gs.fa_bc.fa_co.fa_orf.txt', '-minsize', '200']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/share/tetrimmer/boundarycrop.py", line 654, in find_boundary_and_crop
    if orf_domain_plot_object.run_getorf():
  File "/usr/local/share/tetrimmer/orfdomain.py", line 200, in run_getorf
    prcyan('\n' + e.stderr + '\n')
TypeError: can only concatenate str (not "bytes") to str

```
Do you know what might cause these errors?

Kind regards,
Ting-Hsuan

**Comment by qjiangzhao** [comment]
Hi Ting-Hsuan,

Mnay thanks for your feedback!

Did you bind your input files and output paths like this?

`singularity exec --writable-tmpfs \
--bind <your_path_contain_genome_file>:/genome \
--bind <your_path_contain_input_TE_library_file>:/input \
--bind <your_output_path>:/output \
--bind <your_path_to_store_PFAM_database>:/pfam \  
<your_path_contain_sif_file>/tetrimmer_1.4.0--hdfd78af_0.sif \
TEtrimmer \
-i /input/<TE_library_name.fasta> \
-g /genome/<genome_file_name.fasta> \
-o /output \
--pfam_dir /pfam \
-t 20 --classify_all`

If you had done that, you can run TEtrimmer with "-t 1 --debug" and send me all your output and log files. I will have a further look. 

Yours sincerely
Jiangzhao

**Comment by ting-hsuan-chen** [comment]
Thank you @qjiangzhao for answering my questions via emails and helping me with the test runs. Just posting the solution related to getorf errors to close this issue.

After running '--debug' and checking the outputs, Jiangzhao suggested me to use conda install instead of singularity. 

After installing TEtrimmer via conda install, `TEtrimmer --help` showed "command not found". But I was able to call TEtrimmer by navigating to the TEtrimmer.py in the folder cloned from TEtrimmer github repo. Like the following:
```bash
Python $TEtrimmer_repo/tetrimmer/TEtrimmer.py --input_file $OUT/test/test_input.fa \
--genome_file $OUT/test/test_genome.fasta \
--output_dir $OUT/test \
--pfam_dir $pfamdir/pfam \
--num_threads $THREADS \
--classify_all 
```

The getorf error didn't appear anymore. However, I got an error as same as #26. I followed the suggestion `conda install conda-forge::ghostscript` and then had a successful test run!
